efi/attributes/attributes.go

```diff
@@ -140,3 +143,16 @@ func WriteEfivars(name string, attrs Attributes, b []byte) error {
 	}
 	return nil
 }
+
+// Parse an efivar as a UTF-16 string.
+func ParseUtf16Var(data *bytes.Buffer) (string, error) {
+	utf16 := unicode.UTF16(unicode.LittleEndian, unicode.IgnoreBOM)
+	utf16Reader := transform.NewReader(data, utf16.NewDecoder())
+	b, err := io.ReadAll(utf16Reader)
+
+	if err != nil {
+		return "", err
+	}
+
+	return string(bytes.Trim(b, "\x00")), nil
```

Comment:
I think we should explicitly check that the string is null terminated. Else it's simply not a string from UEFI. Would give a false impression?

I also think this belongs in `efi/util`

Reply:
I've moved this around, but I'm not 100% certain about what we should check: the bytes themselves, or the value after decoding?

On top of that, I was adding a test, and noticed an extra control character in the value returned by this, which does not seem right, but I'm honestly not sure why we're seeing it.

Reply:
Just do a `if b[len(b)-1] != 0` check I reckon and return an error. The second paragraph was cleared up on IRC, but confusion stems from reading the raw files includes the efi attributes :)

Reply:
Done!